universum/submit.py

```diff
@@ -30,8 +30,8 @@ def define_arguments(parser):
                             help="List of vcs or directories to be reconciled for commit. "
                                  "Relative paths starting at client root are supported")
 
-    def __init__(self, *args, **kwargs):
-        super(Submit, self).__init__(*args, **kwargs)
+    def __init__(self, *args, **kwargs) -> None:
```

Comment:
does this belong to the same PR as the change above?

Reply:
Yes. As I mentioned in the description, this is added to illustrate the issue.
Without annotation on the client side, annotations in the gravity have no effect.

Also, I've tried to check if annotations give proper results.
It turned out that at least for vcs_factory() the mypy believes its type is ProjectDirectory.
Any attempt to use the result with real type (SubmitVcs) result it typing errors.
For instance, the following line produces error:
```python
change = self.vcs.driver.submit_new_change(self.settings.commit_message,
                                           path_list,
                                           review=self.settings.review,
                                           edit_only=self.settings.edit_only)
```

Another issue is that there is no possibility to annotate self.vcs with SubmitVcs, because it is a variable: https://mypy.readthedocs.io/en/latest/common_issues.html#variables-vs-type-aliases

Reply:
Got it